lib/IRGen/GenMeta.cpp

```diff
@@ -1226,6 +1206,19 @@ namespace {
       addVTable();
     }
 
+    void addIncompleteMetadataOrRelocationFunction() {
```

Comment:
Should we actually add more fields to it, or have it point to a separate pattern structure in place of the incomplete metadata?

Reply:
Good question.  The only reason we wouldn't want to just replace the IncompleteMetadata with a pattern reference is if there's a situation where we'd want an allocation function but *not* a pattern, and that just seems really unlikely to me.  The reverse scenario — where we want a patten but not an allocation function, e.g. as a load-time optimization — is more plausible but doesn't need to be specially optimized because it's easy to fit a metadata pointer into the pattern — we can just union the allocation function in the pattern with a constant metadata, discriminated by some flag in the pattern (which needs flags anyway).

That does make me wonder, though, whether having a resilient superclass is the right condition to flag the presence of a pattern.  Is there any other reasonable way to store this?

Reply:
We could just add another flag to the type context descriptor flags.

Reply:
Yeah.  Or maybe make IncompleteMetadata a pointer union, since they're both pointers to structures now instead of one side being a pointer to code.